addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/transceiver/EnOceanTransceiver.java

```diff
@@ -100,7 +99,7 @@ private synchronized void send() throws IOException {
 
                             // slowdown sending of message to avoid hickups at receivers
                             // Todo tweak sending intervall (250 ist just a first try)
-                            timeOut = scheduler.schedule(() -> {
+                            scheduler.schedule(() -> {
```

Comment:
You must always store the scheduler future. Because when the class gets destroyed (a dispose() method is missing here?) the future must be canceled.

The reason is that this binding can be stopped / uninstalled / the thing deactivated at any time and all resources must be freed at that point including calling dispose() on any referenced objects of the thing handlers.

Reply:
Ohh I see, just saw that `timeOut` was not used, so I removed it.

Now I have reverted this and call `timeOut.cancel()` in the method `ShutDown()`, and before every new assignment. I hope these are the right places? If not, let me know :-)